projects/fsastorefrontlib/src/cms-components/my-account/quote/quote-details/quote-details.component.ts

```diff
@@ -40,11 +41,13 @@ export class QuoteDetailsComponent implements OnInit, OnDestroy {
       ])
         .pipe(
           map(([routingData, userId]) => {
-            const quoteId = routingData.state.params.quoteId;
-            if (quoteId) {
-              this.quoteService.loadQuoteDetails(quoteId);
+            if (!routingData.nextState) {
+              const quoteId = routingData.state.params.quoteId;
+              if (quoteId) {
+                this.quoteService.loadQuoteDetails(quoteId, userId);
```

Comment:
Also I extended `loadQuoteDetails ` method with additional parameter `userId `, and removed subscribing on the `this.userIdService.getUserId()` from the service.